src/module/JobDetails/JobDetails.tsx

```diff
@@ -0,0 +1,118 @@
+import { useParams } from "react-router";
+import { useContext } from "react";
+
+import "./JobDetails.scss";
+import { JobContext } from "../Job/context/JobContext";
+import { MainButton } from "../../ui/button/MainButton/MainButton";
+
+export const JobDetails = () => {
+  const { jobs } = useContext(JobContext);
+
+  const { id } = useParams();
+
+  const position = jobs.find((p) => p.id === +id!);
```

Comment:
Значит смотри. У тебя неправильно настроен роутер. Также ты не делаешь проверок на валидность твоего id, полученного из параметров. Не показываешь состояние загрузки. В твоем случае, все это надо делать руками. В современном фронтенде, используется современные библиотеки, которые все это делают сами, но пока до них еще далеко. Ниже исправленный код
```suggestion
  const { jobs } = useContext(JobContext);
  const { id } = useParams<{ id: string }>();
  const navigate = useNavigate();

  const position = jobs.find((p) => p.id === Number(id));

  useEffect(() => {
    if (jobs.length && !position) navigate("/");
  }, [jobs, position, navigate]);

  if (!jobs.length) return <p>Loading...</p>;
```